x-pack/plugins/ml/common/types/saved_objects.ts

```diff
@@ -7,18 +7,31 @@
 
 import type { ErrorType } from '../util/errors';
 export type JobType = 'anomaly-detector' | 'data-frame-analytics';
-export const ML_SAVED_OBJECT_TYPE = 'ml-job';
+export type TrainedModelType = 'trained-model';
+
+export const ML_JOB_SAVED_OBJECT_TYPE = 'ml-job';
+export const ML_TRAINED_MODEL_SAVED_OBJECT_TYPE = 'ml-trained-model';
 export const ML_MODULE_SAVED_OBJECT_TYPE = 'ml-module';
+export type MlSavedObjectType =
+  | typeof ML_JOB_SAVED_OBJECT_TYPE
+  | typeof ML_TRAINED_MODEL_SAVED_OBJECT_TYPE
+  | typeof ML_MODULE_SAVED_OBJECT_TYPE;
 
 export interface SavedObjectResult {
-  [jobId: string]: { success: boolean; type: JobType; error?: ErrorType };
+  [id: string]: { success: boolean; type: JobType | TrainedModelType; error?: ErrorType };
 }
 
+export type SyncResult = {
+  [jobType in JobType | TrainedModelType]?: {
```

Comment:
`jobType` is used throughout the ML code base. It needs to change to something like `mlSavedObjectType` but that is a mouthful, so suggestions are welcome.

Also as changing it will be a simple find and replace task, I would like to do it in a specific followup PR to reduce churn.

Reply:
`mlSavedObjectType` sounds a good choice to me